### ISSUE TITLE
Blank lines break `value_from`'s csv/csv2dict formats 

### ISSUE THREAD

**Issue by ajkerrigan** [cloud-custodian/cloud-custodian#9589]
### Describe the bug

**TL;DR:** Blank lines in CSV files lead to empty `value_from` lookups for the `csv` and `csv2dict` formats. The `csv2dict` case emits a warning but otherwise completes successfully, while the `csv` case bombs with an exception.

If I have a CSV file in this format:

```csv
account_id,reason
111111111111,special
222222222222,
444444444444,
```

I can use `value_from` with either the `csv` or `csv2dict` formats to drive account-based exceptions:

```yaml
conditions:
  - type: value
    key: account_id
    op: not-in
    value_from:
      url: s3://my_bucket/account_exclusions.csv
      format: csv
      expr: 0
```

```yaml
conditions:
  - type: value
    key: account_id
    op: not-in
    value_from:
      url: s3://my_bucket/account_exclusions.csv
      format: csv2dict
      expr: account_id
```

If I accidentally have a blank line in that file:

```csv
account_id,reason
111111111111,special
222222222222,

444444444444,
```

It leads to this observable behavior for `csv2dict`:

```console
2024-06-27 08:41:41,348: custodian.commands:DEBUG Loaded file /home/aj/code/sandbox-policies/csv-blank-line-test.yml. Contains 1 policies
2024-06-27 08:41:41,373: custodian.aws:DEBUG using default region:us-east-2 from boto
2024-06-27 08:41:42,172: custodian.resolver:WARNING ValueFrom filter: account_id key returned None
2024-06-27 08:41:42,172: custodian.policy:INFO Skipping policy:csv-blank-line-test due to execution conditions
```

And this for `csv`:

```console
2024-06-27 08:42:03,660: custodian.commands:DEBUG Loaded file /home/aj/code/sandbox-policies/csv-blank-line-test.yml. Contains 1 policies
2024-06-27 08:42:03,685: custodian.aws:DEBUG using default region:us-east-2 from boto
2024-06-27 08:42:04,560: custodian.commands:ERROR Error while executing policy csv-blank-line-test, continuing
Traceback (most recent call last):
  File "/home/aj/code/cloud-custodian/c7n/commands.py", line 308, in run
    policy()
  File "/home/aj/code/cloud-custodian/c7n/policy.py", line 1382, in __call__
    resources = PullMode(self).run()
                ^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/policy.py", line 316, in run
    if not self.policy.is_runnable():
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/policy.py", line 1210, in is_runnable
    return self.conditions.evaluate(event)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/policy.py", line 1130, in evaluate
    state = all([f.process([policy_vars], event) for f in self.filters])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/filters/core.py", line 600, in process
    return super(ValueFilter, self).process(resources, event)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/filters/core.py", line 205, in process
    return list(filter(self, resources))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/filters/core.py", line 587, in __call__
    matched = self.match(i)
              ^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/filters/core.py", line 639, in match
    self.v = values.get_values()
             ^^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/resolver.py", line 180, in get_values
    contents = self._get_values()
               ^^^^^^^^^^^^^^^^^^
  File "/home/aj/code/cloud-custodian/c7n/resolver.py", line 236, in _get_values
    return set([d[self.data['expr']] for d in data])
                ~^^^^^^^^^^^^^^^^^^^
```

### What did you expect to happen?

I _think_ I expect Custodian to skip those empty lines for lookup purposes. Though having a louder/clearer error message about a look failure might be a safer way to go.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Please copy/paste the following info along with any bug reports:

Custodian:   0.9.38
Python:      3.12.2 (main, Mar 10 2024, 10:02:36) [GCC 11.4.0]
Platform:    posix.uname_result(sysname='Linux', nodename='mosa', release='6.8.8-060808-generic', version='#202404271536 SMP PREEMPT_DYNAMIC Sat Apr 27 15:54:17 UTC 2024', machine='x86_64')
Using venv:  True
Docker: False
Installed: 

PyJWT==2.8.0
PyYAML==6.0.1
adal==1.2.7
appdirs==1.4.4
applicationinsights==0.11.10
apscheduler==3.10.4
argcomplete==3.3.0
attrs==23.2.0
azure-common==1.1.28
azure-core==1.30.1
azure-cosmos==3.2.0
azure-cosmosdb-nspkg==2.0.2
azure-cosmosdb-table==1.0.6
azure-functions==1.19.0
azure-graphrbac==0.61.1
azure-identity==1.16.0
azure-keyvault==4.2.0
azure-keyvault-certificates==4.8.0
azure-keyvault-keys==4.9.0
azure-keyvault-secrets==4.8.0
azure-mgmt-advisor==9.0.0
azure-mgmt-apimanagement==1.0.0
azure-mgmt-appconfiguration==0.7.0
azure-mgmt-applicationinsights==1.0.0
azure-mgmt-appplatform==8.0.0
azure-mgmt-authorization==1.0.0
azure-mgmt-automation==0.1.1
azure-mgmt-batch==15.0.0
azure-mgmt-cdn==12.0.0
azure-mgmt-cognitiveservices==11.0.0
azure-mgmt-compute==19.0.0
azure-mgmt-containerinstance==7.0.0
azure-mgmt-containerregistry==10.3.0
azure-mgmt-containerservice==15.1.0
azure-mgmt-core==1.4.0
azure-mgmt-cosmosdb==6.4.0
azure-mgmt-costmanagement==1.0.0
azure-mgmt-databricks==1.0.0b1
azure-mgmt-datafactory==1.1.0
azure-mgmt-datalake-analytics==0.5.0
azure-mgmt-datalake-nspkg==3.0.1
azure-mgmt-datalake-store==0.5.0
azure-mgmt-desktopvirtualization==1.1.0
azure-mgmt-dns==8.0.0b1
azure-mgmt-eventgrid==8.0.0
azure-mgmt-eventhub==11.0.0
azure-mgmt-frontdoor==1.2.0
azure-mgmt-hdinsight==7.0.0
azure-mgmt-iothub==1.0.0
azure-mgmt-keyvault==8.0.0
azure-mgmt-kusto==2.2.0
azure-mgmt-logic==9.0.0
azure-mgmt-machinelearningservices==1.0.0
azure-mgmt-managementgroups==1.0.0b1
azure-mgmt-monitor==2.0.0
azure-mgmt-msi==1.0.0
azure-mgmt-network==17.1.0
azure-mgmt-nspkg==3.0.2
azure-mgmt-policyinsights==1.0.0
azure-mgmt-rdbms==10.1.0
azure-mgmt-recoveryservices==2.5.0
azure-mgmt-recoveryservicesbackup==7.0.0
azure-mgmt-redhatopenshift==1.4.0
azure-mgmt-redis==12.0.0
azure-mgmt-resource==16.1.0
azure-mgmt-resourcegraph==7.0.0
azure-mgmt-search==8.0.0
azure-mgmt-security==1.0.0
azure-mgmt-servicebus==8.2.0
azure-mgmt-servicefabric==1.0.0
azure-mgmt-signalr==0.4.0
azure-mgmt-sql==1.0.0
azure-mgmt-storage==17.1.0
azure-mgmt-streamanalytics==1.0.0
azure-mgmt-subscription==1.0.0
azure-mgmt-synapse==2.0.0
azure-mgmt-trafficmanager==0.51.0
azure-mgmt-web==2.0.0
azure-nspkg==3.0.2
azure-storage-blob==12.20.0
azure-storage-common==2.1.0
azure-storage-file==2.1.0
azure-storage-file-share==12.16.0
azure-storage-queue==12.10.0
boto3==1.34.118
botocore==1.34.118
c7n==0.9.37
cachetools==5.3.3
certifi==2024.6.2
cffi==1.16.0
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
cryptography==42.0.7
decorator==5.1.1
distlib==0.3.8
docutils==0.18.1
dogpile.cache==1.3.3
google-api-core==2.19.0
google-api-python-client==2.131.0
google-auth==2.29.0
google-auth-httplib2==0.2.0
google-cloud-appengine-logging==1.4.3
google-cloud-audit-log==0.2.5
google-cloud-core==2.4.1
google-cloud-logging==3.10.0
google-cloud-monitoring==2.21.0
google-cloud-storage==2.16.0
google-crc32c==1.5.0
google-resumable-media==2.7.0
googleapis-common-protos==1.63.1
grpc-google-iam-v1==0.13.0
grpcio==1.64.1
httplib2==0.22.0
idna==3.7
importlib-metadata==6.11.0
iso8601==2.1.0
isodate==0.6.1
jmespath==1.0.1
jsonpatch==1.33
jsonpointer==2.4
jsonschema==4.22.0
jsonschema-specifications==2023.12.1
keystoneauth1==5.6.0
kubernetes==29.0.0
msal==1.28.0
msal-extensions==1.1.0
msrest==0.7.1
msrestazure==0.6.4.post1
munch==4.0.0
netaddr==0.7.20
netifaces==0.11.0
oauthlib==3.2.2
openstacksdk==0.52.0
os-service-types==1.7.0
packaging==24.0
pbr==6.0.0
portalocker==2.8.2
proto-plus==1.23.0
protobuf==4.25.3
pyasn1==0.6.0
pyasn1-modules==0.4.0
pycparser==2.22
pyparsing==3.1.2
pyrate-limiter==2.10.0
python-dateutil==2.9.0.post0
pytz==2024.1
pyyaml==6.0.1
referencing==0.31.1
requests==2.32.3
requests-oauthlib==2.0.0
requestsexceptions==1.4.0
retrying==1.3.4
rpds-py==0.18.1
rsa==4.9
s3transfer==0.10.1
six==1.16.0
stevedore==5.2.0
tabulate==0.9.0
typing-extensions==4.12.1
tzlocal==5.2
uritemplate==4.1.1
urllib3==1.26.18
websocket-client==1.8.0
zipp==3.19.1
```

### Policy

```shell
policies:
  - name: csv-blank-line-test
    resource: aws.ec2
    conditions:
      - type: value
        key: account_id
        op: not-in
        value_from:
          url: s3://my-bucket/account_exclusions.csv
          format: csv
          expr: 0

      - type: value
        key: account_id
        op: not-in
        value_from:
          url: s3://my-bucket/account_exclusions.csv
          format: csv2dict
          expr: account_id
```

### Extra information or context

I believe the `csv2dict` behavior comes from [this zip() operation](https://github.com/cloud-custodian/cloud-custodian/blob/0397dd4a97c0b56479247bb6a7a85f196f7d381e/c7n/resolver.py#L228) not producing any data, but also not raising an error.
